core/src/main/java/org/apache/iceberg/jdbc/JdbcTable.java

```diff
@@ -0,0 +1,187 @@
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JdbcTable {
+  public static final String SQL_TABLE_NAME = "iceberg_tables";
+  public static final String SQL_TABLE_DDL =
+          "CREATE TABLE " + JdbcTable.SQL_TABLE_NAME +
+                  "(catalog_name VARCHAR(1255) NOT NULL," +
+                  "table_namespace VARCHAR(1255) NOT NULL," +
+                  "table_name VARCHAR(1255) NOT NULL," +
+                  "metadata_location VARCHAR(32768)," +
+                  "previous_metadata_location VARCHAR(32768)," +
+                  "PRIMARY KEY (catalog_name, table_namespace, table_name)" +
+                  ")";
+  public static final String SQL_SELECT = "SELECT * FROM " + SQL_TABLE_NAME +
+          " WHERE catalog_name = ? AND table_namespace = ? AND table_name = ? ";
+  public static final String SQL_SELECT_ALL = "SELECT * FROM " + SQL_TABLE_NAME +
+          " WHERE catalog_name = ? AND table_namespace = ?";
+  public static final String SQL_UPDATE_METADATA_LOCATION = "UPDATE " + SQL_TABLE_NAME +
+          " SET metadata_location = ? , previous_metadata_location = ? " +
+          " WHERE catalog_name = ? AND table_namespace = ? AND table_name = ? AND metadata_location = ?";
+  public static final String SQL_INSERT = "INSERT INTO " + SQL_TABLE_NAME +
+          " (catalog_name, table_namespace, table_name, metadata_location, previous_metadata_location) " +
+          " VALUES (?,?,?,?,?)";
+  public static final String SQL_UPDATE_TABLE_NAME = "UPDATE " + SQL_TABLE_NAME +
+          " SET table_namespace = ? , table_name = ? " +
+          " WHERE catalog_name = ? AND table_namespace = ? AND table_name = ? ";
+  public static final String SQL_DELETE = "DELETE FROM " + SQL_TABLE_NAME +
+          " WHERE catalog_name = ? AND table_namespace = ? AND table_name = ? ";
+
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcTable.class);
+  private final String catalogName;
+  private final Connection dbConn;
+
+  public JdbcTable(Connection dbConn, String catalogName) {
+    this.dbConn = dbConn;
+    this.catalogName = catalogName;
+  }
+
+  public void setAutoCommitOff() throws SQLException {
+    this.dbConn.setAutoCommit(false);
+  }
+
+  public void setAutoCommitOn() throws SQLException {
+    this.dbConn.setAutoCommit(true);
+  }
+
+  public void commit() throws SQLException {
+    this.dbConn.commit();
+  }
+
+  public void rollback() throws SQLException {
+    this.dbConn.rollback();
+  }
+
+  public boolean isExists(TableIdentifier tableIdentifier) {
```

Comment:
We would normally omit the "is" from a method name if it is clear that the return value is a boolean. I think that `exists` meets that requirement because it is natural to say "if table exists" instead of "if table is exists".